src/openms/source/ANALYSIS/OPENSWATH/PeakPickerMobilogram.cpp

```diff
@@ -0,0 +1,318 @@
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Justin Sing $
+// $Authors: Justin Sing $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/ANALYSIS/OPENSWATH/PeakPickerMobilogram.h>
+#include <iostream>
+#include <iomanip>    // For std::setw
+#include <vector>
+#include <algorithm>  // For std::min_element and std::max_element
+#include <fstream>
+#include <iostream>
+#include <random>
+#include <sstream>
+
+namespace OpenMS
+{
+    PeakPickerMobilogram::PeakPickerMobilogram() :
+            DefaultParamHandler("PeakPickerMobilogram")
+//            ProgressLogger()
+    {
+        defaults_.setValue("sgolay_frame_length", 9, "The number of subsequent data points used for smoothing.\nThis number has to be uneven. If it is not, 1 will be added.");
+        defaults_.setValue("sgolay_polynomial_order", 3, "Order of the polynomial that is fitted.");
+        defaults_.setValue("gauss_width", 0.002, "Gaussian width in seconds, estimated peak size.");
+        defaults_.setValue("use_gauss", "false", "Use Gaussian filter for smoothing (alternative is Savitzky-Golay filter)");
+        defaults_.setValidStrings("use_gauss", {"false","true"});
+
+        defaults_.setValue("peak_width", -1.0, "Force a certain minimal peak_width on the data (e.g. extend the peak at least by this amount on both sides) in seconds. -1 turns this feature off.");
+        defaults_.setValue("signal_to_noise", 1.0, "Signal-to-noise threshold at which a peak will not be extended any more. Note that setting this too high (e.g. 1.0) can lead to peaks whose flanks are not fully captured.");
+        defaults_.setMinFloat("signal_to_noise", 0.0);
+
+        defaults_.setValue("sn_win_len", 1, "Signal to noise window length.");
+        defaults_.setValue("sn_bin_count", 4, "Signal to noise bin count.");
+        defaults_.setValue("write_sn_log_messages", "false", "Write out log messages of the signal-to-noise estimator in case of sparse windows or median in rightmost histogram bin");
+        defaults_.setValidStrings("write_sn_log_messages", {"true","false"});
+
+        defaults_.setValue("remove_overlapping_peaks", "false", "Try to remove overlapping peaks during peak picking");
+        defaults_.setValidStrings("remove_overlapping_peaks", {"false","true"});
+
+        defaults_.setValue("method", "corrected", "Which method to choose for mobilogram peak-picking (OpenSWATH legacy on raw data, corrected picking on smoothed mobilogram).");
+        defaults_.setValidStrings("method", {"legacy","corrected","crawdad"});
+
+        // write defaults into Param object param_
+        defaultsToParam_();
+        updateMembers_();
+
+        // PeakPickerHiRes pp_;
+        Param pepi_param = pp_.getDefaults();
+        pepi_param.setValue("signal_to_noise", signal_to_noise_);
+        // disable spacing constraints, since we're dealing with chromatograms
```

Comment:
maybe double check if all use of "chromatograms" have been replaced with "mobilograms"